google-cloud-storage/acceptance/storage_helper.rb

```diff
@@ -105,20 +105,31 @@ def self.run_one_method klass, method_name, reporter
   end
 end
 
-def safe_gcs_execute retries: 20, delay: 2
+def safe_gcs_execute retries: 8
   current_retries = 0
   loop do
     begin
       return yield
     rescue Google::Cloud::ResourceExhaustedError
       raise unless current_retries >= retries
-
-      sleep delay
+      
       current_retries += 1
+      sleep get_retry_delay(current_retries)
+      # random_adder = rand(0.0..3.0)
+      # sleep (2 ** current_retries) + random_adder
+      # max_sleep_seconds = Float(2 ** current_retries)
+      # sleep rand(0..max_sleep_seconds)
     end
   end
 end
 
+def get_retry_delay current_retries, delay_multiplier: 3
+  random_adder = rand(0.0..3.0)
+  wait_time = (delay_multiplier ** current_retries) + 1 + random_adder
+  max_allowed_delay = 75.0
+  [wait_time, max_allowed_delay].min
```

Comment:
The random adder should be added _after_ the delay is capped by max_allowed_delay. Otherwise, once the max is reached (currently after only 3 iterations), the random_adder will always be suppressed by this cap, and we lose the benefit of the jitter.